src/Storages/System/StorageSystemDetachedParts.cpp

```diff
@@ -50,11 +80,14 @@ Pipe StorageSystemDetachedParts::read(
         for (const auto & p : parts)
         {
             size_t i = 0;
+            String detached_part_path = fs::path(MergeTreeData::DETACHED_DIR_NAME) / p.dir_name;
             new_columns[i++]->insert(info.database);
             new_columns[i++]->insert(info.table);
             new_columns[i++]->insert(p.valid_name ? p.partition_id : Field());
             new_columns[i++]->insert(p.dir_name);
+            new_columns[i++]->insert(calculateTotalSizeOnDisk(p.disk, fs::path(info.data->getRelativeDataPath()) / detached_part_path));
```

Comment:
It may slow down `select ... from system.detached_parts` significantly, especially if object storage is involved. Moreover, a query slows down even if the `bytes_on_disk` column is not queried. We should not calculate `bytes_on_disk` if it was not queried.